### DISCUSSION TITLE
Theme overline does not appear over org-modern elements

### DISCUSSION THREAD

**Discussion by spencerjackson** [minad/org-modern#248] [Q&A]
Hello! First off, thank you for making such a wonderful package!

I've recently switched to the modus-vivendi theme, and configured it to show overlines above org headlines. Unfortunately, this appears to conflict with org-modern, and the overline will not appear over prettified elements:

![image](https://github.com/user-attachments/assets/8c4029cf-000c-4af5-b066-b7bc7e7d5873)

Looking at the text in the headline, it appears to set with faces like `(org-headline-done org-level-2)`. `org-level-2` inherits from `modus-themes-heading-2`, which sets a depth-specific overline color. Looking at the text in a prettified DONE tag, I see that the only face it possesses is `org-modern-done`.

I considered adding overline attributes to the org-modern faces, but the correct overline color depends on the depth of the headline.

I think a way to fix this might be to prepend the org-modern faces to the pre-existing set of faces existing on the text using `add-face-text-property` instead of `put-text-property.` Then, org-modern would override any face attributes set by the theme, but still allow the theme to provide defaults.

**Comment by minad** [comment]
Hi! Thanks for the report. Can you provide a patch, which demonstrates the fix via add-face-text-property and provide a minimal configuration I could use for testing? It could however be that this cannot be fixed satisfactorily since the TODO and tag boxes will be in conflict with the overlines. In your screenshot the boxes are overly large which is not the intended design of org-modern. org-modern itself is opionated and it can conflict with other special settings like headline overlines.

**Reply by spencerjackson** [reply]
Sure thing! I've submitted PR #249 with the patch, repro configuration, and new screenshots. And yeah, totally understand that not every conflict between org-modern and arbitrary themes can be arbitrated. Thanks for looking at it!

**Reply by minad** [reply]
Thanks I will try it. Do you have FSF copyright assignment which is needed to merge this?

**Reply by minad** [reply]
By looking at the screenshots, they look a little differently (font size, colors). The borders are fixed however. But this makes me a bit wary about this change. As I wrote Org-modern may not handle this too well - it wasn't designed to handle mixes of theme and Org-modern styling. I also explicitly asked Prot to not override Org-modern via his themes to avoid interference, since this has led to problems before.
